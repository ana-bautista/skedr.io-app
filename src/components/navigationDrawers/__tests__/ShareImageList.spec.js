import { shallowMount, mount } from '@vue/test-utils'
import comp from '../ShareImageList.vue'
import Vue from 'vue'
import Vuex from 'vuex'
import Vuetify from 'vuetify'

Vue.use(Vuetify)
Vue.use(Vuex)

describe('ShareImageList', () => {
  it('renders a list element', () => {
    const wrapper = shallowMount(comp, {
      propsData: { photo: { photoId: 'photoId', title: 'title', urlSq: 'urlSq' } }
    })
    expect(wrapper).toMatchSnapshot()
  })

  it('is a public photo', () => {
    const wrapper = shallowMount(comp, {
      propsData: { photo: { isPublic: true } }
    })
    expect(wrapper.vm.isPrivate).toBe(false)
  })

  it('removes the photo from the pool', () => {
    const actions = { 'pool/remove': jest.fn() }
    const wrapper = mount(comp, {
      propsData: { photo: { photoId: 'photoId' } },
      store: new Vuex.Store({
        actions,
        modules: {
          namespaced: true,
          pool: {}
        }
      })
    })
    const result = wrapper.find('button').trigger('click')
    expect(actions['pool/remove']).toHaveBeenCalled()
  })
})
