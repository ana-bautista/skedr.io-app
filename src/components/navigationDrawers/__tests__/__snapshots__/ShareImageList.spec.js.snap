// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShareImageList renders a list element 1`] = `
<v-list-tile-stub activeclass="primary--text" avatar="true">
  <v-list-tile-avatar-stub size="40">
    <v-badge-stub color="black" value="true" transition="fab-transition" overlap="true"> <img src="urlSq"></v-badge-stub>
  </v-list-tile-avatar-stub>
  <v-list-tile-content-stub>
    <v-list-tile-title-stub>title</v-list-tile-title-stub>
  </v-list-tile-content-stub>
  <v-list-tile-action-stub>
    <v-btn-stub activeclass="v-btn--active" ripple="true" tag="button" icon="true" type="button">
      <v-icon-stub>close</v-icon-stub>
    </v-btn-stub>
  </v-list-tile-action-stub>
</v-list-tile-stub>
`;
